ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/individual-journey/components/video-view-base/video-view-base.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable, ViewChild, OnInit } from '@angular/core';
+import { IndividualBaseComponent } from '../../pages/individual-base-component/individual-base.component';
+import { IndividualJourney } from '../../individual-journey';
+import { VideoViewComponent } from '../../components/video-view/video-view.component';
+import { VideoUrlService } from '../../services/video-url.service';
+import { VideoFiles } from '../../services/video-files';
+
+@Injectable()
+export class VideoViewBaseComponent extends IndividualBaseComponent implements OnInit {
+
+  constructor(journey: IndividualJourney, private videoUrlService: VideoUrlService) {
+    super(journey);
+  }
+
+  @ViewChild(VideoViewComponent)
+  videoViewComponent: VideoViewComponent;
+
+  videoSource: string;
+  disabledReplay = true;
+  videoFile: VideoFiles;
+
+  ngOnInit() {
+    this.videoSource = this.videoUrlService.getVideoFileUrl(this.videoFile);
```

Comment:
so this piece of code here means that the child component needs to know about the internals of when the base component needs the `videoFile` to be set. Another option would be to have the child component do this call as a part of calling the parent controller constructor

```typescript
// in child class
constructor(journey: IndividualJourney, private videoUrlService: VideoUrlService) {
    super(journey, videoeUrlService.getVideoFileUrl(VideoFiles.BeforeTheDay));
}

// and then the constructor in this class would be more like
  constructor(journey: IndividualJourney, private string: videoUrl) {
    super(journey);
    this.videoUrl = videoUrl;
  }
```

that way the child component and base class get more separated in terms of functionality and dependencies.

Reply:
Personally, I am not very sure about calling method in super(…), does not look right

Reply:
No that still looks good. Now the child component doesn't need to know about the base and the tests are reused so that looks great to me, nice work!